homeassistant/components/home_connect/light.py

```diff
@@ -84,14 +85,58 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Home Connect light."""
+    known_appliance_ids: set[str] = set()
 
-    async_add_entities(
-        [
+    def get_entities_for_appliance(
+        appliance: HomeConnectApplianceData,
+    ) -> list[HomeConnectLight]:
+        """Get a list of entities."""
+        known_appliance_ids.add(appliance.info.ha_id)
+        return [
             HomeConnectLight(entry.runtime_data, appliance, description)
             for description in LIGHTS
-            for appliance in entry.runtime_data.data.values()
             if description.key in appliance.settings
-        ],
+        ]
+
+    async_add_entities(
+        entity
+        for appliance in entry.runtime_data.data.values()
+        for entity in get_entities_for_appliance(appliance)
+    )
+
+    def handle_paired_or_connected_appliance() -> None:
+        """Handle new paired appliance or a appliance that has been connected."""
+        for appliance in entry.runtime_data.data.values():
+            if appliance.info.ha_id in known_appliance_ids:
+                entity_registry = er.async_get(hass)
+                async_add_entities(
```

Comment:
If we've added the `ha_id` to `known_appliance_ids` we've already created and added entities for that appliance. Then we shouldn't add those entities again.

What is the end result we want to achieve here?

Reply:
Sorry, that change I made for `known_appliance_ids` wasn't very well thought, I did it first by saving the unique id of the entities, and I reverted the change already.
And the point of this is that some appliances doesn't report data while turned off because AFAIK they are disconnected, so when they get connected again, we can check if there's new entities to add.

Reply:
Ok. We need to add code comments or docstrings explaining this. 👍